src/modes/shop/shop.cpp

```diff
@@ -1070,6 +1070,33 @@ ShopMode::~ShopMode()
 {
     _SaveShopData();
 
+    for (auto item : _available_buy) {
```

Comment:
The pointers within the maps

```
_available_buy
_available_sell
_available_trade
```
were not being cleaned up.  I added some code to free their ShopObjects.

These maps are cleared in multiple places.  So, the code is a little messy.